mico-core/src/main/java/io/github/ust/mico/core/dto/MicoApplicationWithServicesDTO.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.github.ust.mico.core.dto;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+import io.github.ust.mico.core.configuration.extension.CustomOpenApiExtentionsPlugin;
+import io.github.ust.mico.core.dto.MicoApplicationDTO.MicoApplicationDeploymentStatus;
+import io.github.ust.mico.core.model.MicoApplication;
+import io.github.ust.mico.core.model.MicoService;
+import io.swagger.annotations.ApiModelProperty;
+import io.swagger.annotations.Extension;
+import io.swagger.annotations.ExtensionProperty;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.Accessors;
+
+/**
+ * DTO for a {@link MicoApplication} including all of
+ * its associated {@link MicoService MicoServices}.
+ */
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Accessors(chain = true)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(Include.NON_NULL)
+public class MicoApplicationWithServicesDTO {
+
+    /**
+     * The {@link MicoApplication}.
+     */
+    @ApiModelProperty(extensions = {@Extension(
+            name = CustomOpenApiExtentionsPlugin.X_MICO_CUSTOM_EXTENSION,
+            properties = {
+                @ExtensionProperty(name = "title", value = "Application"),
+                @ExtensionProperty(name = "x-order", value = "10"),
+                @ExtensionProperty(name = "description", value = "The application.")
+            }
+        )})
+    private MicoApplicationDTO application;
```

Comment:
Having an application object in a wrapper object is uncomfortable. Using inheritance would be preferable.

Reply:
Done, `MicoApplicationWithServicesDTO` now inherits from `MicoApplicationDTO`.